rest/blip_api_test.go

```diff
@@ -2485,7 +2485,258 @@ func TestBlipDeltaSyncPullRevCache(t *testing.T) {
 
 	assert.Equal(t, deltaCacheHits+1, updatedDeltaCacheHits)
 	assert.Equal(t, deltaCacheMisses, updatedDeltaCacheMisses)
+}
+
+// Check removal handling for unavailable multi-channel revisions in delta
+// sync when fromRevision is a channel removal.
+func TestBlipDeltaSyncPullRevCacheChannelFromRevRemoval(t *testing.T) {
+	if !base.IsEnterpriseEdition() {
+		t.Skipf("Skipping enterprise-only delta sync test.")
+	}
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+
+	sgUseDeltas := base.IsEnterpriseEdition()
+	rtConfig := RestTesterConfig{
+		DatabaseConfig: &DbConfig{
+			DeltaSync: &DeltaSyncConfig{
+				Enabled: &sgUseDeltas,
+			},
+		},
+	}
+	rt := NewRestTester(t, &rtConfig)
+	defer rt.Close()
+
+	// waitForChangesInChannel asserts that waiting for a number of changes in the specified set of channel(s)
+	// was successful.
+	waitForChangesInChannel := func(channels []string, numChanges int) {
+		changesURL := "/db/_changes?filter=sync_gateway/bychannel&channels=" + strings.Join(channels[:], ",")
+		changes, err := rt.WaitForChanges(numChanges, changesURL, "", true)
+		require.NoError(t, err, "Error retrieving changes results")
+		require.Len(t, changes.Results, numChanges)
+	}
+
+	// Create client with a user who has access to channel1.
+	client1, err := NewBlipTesterClientOptsWithRT(t, rt, &BlipTesterClientOpts{
```

Comment:
Can remove client1 in this test, it's not doing anything different than client2.

Reply:
Sure, but this test has been replaced with both TestDeltaSyncWhenFromRevIsChannelRemoval and TestDeltaSyncWhenToRevIsChannelRemoval to check the delta sync behavior.